### PR TITLE
refactor(db): simplify assignments and const export

### PR DESCRIPTION
**Pull request by kripod**
### What are the changes and their implications?

Simplify code and type-check `globalThis`.

### Checklist

- [ ] Tests added for changes
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

File: packages/generator/templates/app/db/index.ts [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+function getPersistentPrismaClient() {
+  // Ensure the Prisma instance is re-used during hot-reloading
+  // Otherwise, a new client would be created on every reload
+  globalThis.prisma ||= new PrismaClient()
+  return globalThis.prisma
 }
 
-export default prisma
+export default process.env.NODE_ENV === "production"
+  ? getPersistentPrismaClient()
```
Comment (BjoernRave) [inline packages/generator/templates/app/db/index.ts:16]: shouldn't this be the other way round?

Comment (kripod) [reply packages/generator/templates/app/db/index.ts:16]: Oh, it should, I’m fixing it!

**Comment by flybayer** [comment]
Thanks @kripod. However, I think we're going to maybe remove the globalThis stuff all together with #1374.

**Comment by kripod** [comment]
@flybayer Sounds great, I didn’t know about that. Thank you for the heads up, I’m closing this for now.

**Comment by flybayer** [comment]
I'm going to re-open, because I think we may keep this after all. Will wait to merge until I see what happens with #1374

**Comment by flybayer** [comment]
@kripod can you fix the failing test?

**Comment by kripod** [comment]
@flybayer I'm not quite sure why the CI run is failing, as it seems to throw a `MaxListenersExceededWarning: Possible EventEmitter memory leak` error without explanation. I would appreciate if you could help debugging the issue.

**Comment by flybayer** [comment]
@kripod those are warnings that you can ignore. The actual test failure is this:

```
@blitzjs/cli: FAIL test/commands/new.test.ts
@blitzjs/cli:   ● Console
@blitzjs/cli:     console.error ../display/dist/display.cjs.development.js:73
@blitzjs/cli:       ✕ Error generating db/index.ts
@blitzjs/cli:     console.error ../display/dist/display.cjs.development.js:73
@blitzjs/cli:       ✕ Error generating db/index.ts
@blitzjs/cli:     console.error ../display/dist/display.cjs.development.js:73
@blitzjs/cli:       ✕ Error generating db/index.ts
@blitzjs/cli:   ● `new` command › when scaffolding new project › pins Blitz to the current version
@blitzjs/cli:     SyntaxError: This experimental syntax requires enabling the parser plugin: 'logicalAssignment' (11:20)
@blitzjs/cli:       at Object._raise (../../node_modules/@babel/parser/src/parser/error.js:60:45)
@blitzjs/cli:       at Object.raiseWithData (../../node_modules/@babel/parser/src/parser/error.js:55:17)
@blitzjs/cli:       at Object.expectPlugin (../../node_modules/@babel/parser/src/parser/util.js:154:18)
@blitzjs/cli:       at Object.parseMaybeAssign (../../node_modules/@babel/parser/src/parser/expression.js:229:14)
@blitzjs/cli:       at Object.parseMaybeAssign (../../node_modules/@babel/parser/src/plugins/typescript/index.js:2347:22)
@blitzjs/cli:       at Object.parseExpression (../../node_modules/@babel/parser/src/parser/expression.js:162:23)
@blitzjs/cli:       at Object.parseStatementContent (../../node_modules/@babel/parser/src/parser/statement.js:284:23)
@blitzjs/cli:       at Object.parseStatementContent (../../node_modules/@babel/parser/src/plugins/typescript/index.js:1979:20)
@blitzjs/cli:       at Object.parseStatement (../../node_modules/@babel/parser/src/parser/statement.js:149:17)
@blitzjs/cli:       at Object.parseBlockOrModuleBlockBody (../../node_modules/@babel/parser/src/parser/statement.js:861:25)
```

which is related to your PR changes.

**Comment by kripod** [comment]
Thanks for your help! Unfortunately, I may not have time to work on this in the near future.

**Comment by Flavyoo** [comment]
@flybayer This is marked as done, but it looks like there is still some things to refine with this PR. Do you know if this is good to close?

**Comment by flybayer** [comment]
This is obsolete because of #1720, #1762

**Pull request closed without merging**